src/genericComponents/Datepicker.tsx

```diff
@@ -0,0 +1,17 @@
+import React, { useState } from 'react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+const CAGDatePicker = (props: any) => {
+  const [selectedDate, setSelectedDate] = useState(new Date());
```

Comment:
Again, this state is only being set internally to this component, so it doesn't serve any purpose for where the component is eventually used. You can remove this as well